src/utils.ts

```diff
@@ -50,14 +52,20 @@ export const getShouldAlertAboutPartyHalfWay = async (
     return false;
   }
 
-  // todo:
-  // 1. create api endpoint that does the following:
-  //   * if partybid - fetch auction reserve price
-  //   * if partybuy - fetch maxPrice from contract
-  //   * if collectionbuy - fetch current floor price
-  // 2. compare (fetched value) / totalContributions >= .5
-  // 3. return boolean
-  const isPartyHalfway = false;
+  const API_ENDPOINT = ""; // todo: add this
+  const totalEthNeeded = await axios.get<{ totalEthNeeded: string }>(
+    `${API_ENDPOINT}?address=${event.party.partyAddress}&type=${event.party.partyType}`
+  );
+
+  // convert wei values to BigNumbers
+  const totalEthNeededBN = BigNumber.from(totalEthNeeded);
+  const totalContributionAmountBN = BigNumber.from(
+    event.contribution.totalAmountContributedToParyInWei
+  );
+
+  // BN.js cannot handle decimal values, so instead we multiply the total contributions by 2 and compare to
+  // total eth needed
+  const isPartyHalfway = totalContributionAmountBN.mul(2).gte(totalEthNeededBN);
```

Comment:
is this true?

Reply:
yes bn.js doesnt support decimals

Reply:
`BN.js` doesnt but `BigNumber` does --- this is a BigNumber here so it actually can support decimal values

Reply:
we should at least remove the comment here